tests/platform_tests/mellanox/test_check_sfp_presence.py

```diff
@@ -10,6 +10,7 @@
 
 pytestmark = [
     pytest.mark.asic('mellanox'),
+    pytest.mark.usefixtures('check_dut_asic_type'),
```

Comment:
I like the idea but it seems less than ideal that we have to add a marker and a fixture anywhere we want to use this check.

This could get really unfriendly if we have a file where the markers are applied to individual methods instead of the whole file because then the author will need to remember to add both for each ASIC-specific test case. Can we somehow incorporate this `check_dut_asic_type` into the `asic` marker itself? Have you tried to see if there is a significant performance impact if we make this `autouse`?

@yxieca @neethajohn any ideas/input here?

Reply:
Thanks Danny. 
Adding two lines for checking ASIC type is indeed annoying.  Actually, I tried to find a way to incorporate the ASIC type as an argument or something into ```check_dut_asic_type``` fixture, but didn't get one. Makeing ```check_dut_asic_type ``` fixture ```autouse``` is another choice. But we must make sure that this fixture is ``` not seen``` by other cases. After all, most cases in our framework needn't this check. So, I'd to get more ideas to make it more ideal.

Reply:
Can test invoke fixture with a supported ASIC list?

Reply:
> Can test invoke fixture with a supported ASIC list?
Thanks Ying. Could you give an example? I don't really understand.

Reply:
I was thinking something like ValidateAsicType(duthost, supported_list), so the supported_list can be used to check against the duthost returned asic type.

Reply:
Thanks Ying. I was thinking some updates that are family with yours. I planed to add a series of fixtures like 
```
only_run_on_broadcom()
only_run_on_mellanox()
only_run_on_xxx()
``` 
So we only call one of the fixtures above when writing test cases for a certain platform. It might seems a little redundant, but more friendly and easier for contributors.

Reply:
> Thanks Ying. I was thinking some updates that are similar with yours. I planed to add a series of fixtures like
> 
> ```
> only_run_on_broadcom()
> only_run_on_mellanox()
> only_run_on_xxx()
> ```
> 
> So we only call one of the fixtures above when writing test cases for a certain platform. It might seems a little redundant, but more friendly and easier for contributors.


Sorry. This may not be a good chioce because a case may run on several platforms, and the above method can't support it well.